src/screens/Login/indexSave.tsx

```diff
@@ -0,0 +1,217 @@
+import { useForm, Controller } from 'react-hook-form';
+import { useNavigation } from '@react-navigation/native';
+import { useLoginMutation, AuthUserInput, LoginDocument } from "../../generated/graphql";
+import { Alert, Text, TouchableOpacity } from 'react-native';
+import Register from '../Register/index';
+
+interface FormProps {
+  email: string;
+  password: string;
+}
+
+function Login() {
+  const { control, handleSubmit, formState: { errors } } = useForm<FormProps>();
+  const [login, {data, loading, error}] = useLoginMutation();
+  const navigation = useNavigation();
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+  const passwordPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/;
+
+  const onSubmit = async (data: AuthUserInput) => {
+
+    const res = await login({
+      variables: {
+        input: {
+          email: data.email,
+          password: data.password,
+        }
+      },
+    });
+    
+    
+  }
+  const handleRegister= () => {
+    navigation.navigate("Register"); // Replace 'Signup' with the name of your signup screen or route
+  };
+  return (
+    <Center flex={1} py="10">
+      <Box  >
+      <Heading size="lg" fontWeight="600" color="coolGray.800" _dark={{
+        color: "warmGray.50"
+      }}>
+        Welcome to CraftPile!
+        </Heading>
+        <Heading mt="1" _dark={{
+        color: "warmGray.200"
+      }} color="coolGray.600" fontWeight="medium" size="xs">
+          Sign in to continue!
+        </Heading>
+      <VStack space={3} mt="5">
+      <FormControl isInvalid={!!errors.email?.message}>
+        <FormControl.Label>Email ID</FormControl.Label>
+        <Controller
+          control={control}
+          
+          name="email"
+          render={({ field: { onChange, onBlur, value } }) => (
+            <Input autoCapitalize="none" onChangeText={onChange} onBlur={onBlur} value={value} />
+          )}
+          rules={{
+            required: { value: true, message: "Email is required" },
+            pattern: { value: emailPattern, message: "Invalid email address" }
+          }}
+        />
+        {errors.email && <FormControl.ErrorMessage>{errors.email.message}</FormControl.ErrorMessage>}
+      </FormControl>
+      <FormControl isInvalid={!!errors.password?.message}>
+        <FormControl.Label>Password</FormControl.Label>
+        <Controller
+          control={control}
+          name="password"
+          render={({ field: { onChange, onBlur, value } }) => (
+            <Input onChangeText={onChange} onBlur={onBlur} value={value} secureTextEntry />
+          )}
+          rules={{
+            required: { value: true, message: "Password is required" },
+            pattern: { value: passwordPattern, message: "Password must contain at least 8 characters, including at least one uppercase letter, one lowercase letter, and one number" }
```

Comment:
for login, we dont really need to apply validation to their password. We already enforced that in registration, all we really care about is that the field is not blank, and of course that the email/pass is a valid login.